orttraining/orttraining/test/gradient/gradient_ops_test.cc

```diff
@@ -1492,8 +1492,12 @@ void TestDropoutGradOp(float ratio, TensorShape& x_shape, bool default_ratio = t
                                                   true, false, true, false});
   if (!default_ratio) {
     test.AddInput<float>("ratio", {1}, ratio_data);
+  } else {
+    test.AddMissingOptionalInput<bool>();
```

Comment:
i might be missing something, just curious, why is the type bool?

Reply:
it is float, actually. the latest version have this fix